src/widgetastic_patternfly/__init__.py

```diff
@@ -374,14 +374,15 @@ def type(self):
 
 class NavDropdown(Widget, ClickableMixin):
     """The dropdowns used eg. in navigation. Usually located in the top navbar."""
-    EXPAND_LOCATOR = './a/span[contains(@class, "caret")]'
+    EXPAND_LOCATOR = ('./a["aria-expanded" and '
+                      '"aria-haspopup" and '
+                      'contains(@class, "dropdown-toggle")]')
 
-    def __init__(self, parent, locator, logger=None):
-        Widget.__init__(self, parent, logger=logger)
-        self.locator = locator
+    ROOT = ParametrizedLocator('//nav//li[//a[@id="{@id}"] and contains(@class, "dropdown"")]')
```

Comment:
I see an extra double quote.

Reply:
Query: I'm guessing now we only need to mention the dropdown ID instead of the whole locator or was that the case before as well?

Reply:
Before it was just taking a locator string, now its defaulting to this locator via ROOT, and the user only has to pass the id for the dropdown.

If their app doesn't have an id for the dropdown, they can subclass and set ROOT as needed - as I've done in integration_tests.

Reply:
Okay! I just saw your PR [#9084](https://github.com/ManageIQ/integration_tests/pull/9084) that you raised in integration_tests. Thanks!